permissions: add permission management for clipboard

littleblack111 opened the pull request:
<!--
BEFORE you submit your PR, please check out the PR guidelines
on our wiki: https://wiki.hyprland.org/Contributing-and-Debugging/PR-Guidelines/
-->


#### Describe your PR, what does it fix/add?
fixes https://github.com/hyprwm/Hyprland/issues/10654


#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
n/a


#### Is it ready for merging, or does it need work?
ready

littleblack111 committed:
clip

littleblack111 committed:
clang said it redundant so..

littleblack111 committed:
idk

vaxerski reviewed (commented):
IMO clip should be enabled-by-default. (Like keyboards)

Also add it to the default config.

Furthermore, I can think of some gaps: what about clipboard managers? I don't think you handled the zwlr clipboard protocol.

littleblack111 committed:
default

littleblack111 committed:
fix: use correct log levels for clipboard permission messages

Use Log::INFO instead of Log::LOG (which doesn't exist) in DataDeviceWlr.cpp
and DataDevice.cpp. Also use the new Log::logger->log() style in XWM.cpp
instead of the deprecated Debug::log().

littleblack111 committed:
fix: address copilot review comments

- Fix log level from INFO to DEBUG in DataDevice.cpp for offer operations
- Remove unused permissionToHumanString function
- Add I18n keys for clipboard read/write permissions
- Add primary selection permission check in XWM.cpp (was only checking clipboard)
- Add async permission handling in DataDeviceWlr.cpp for zwlr protocol
